shared/src/main/scala/mlscript/NuTypeDefs.scala

```diff
@@ -1127,6 +1127,13 @@ class NuTypeDefs extends ConstraintSolver { self: Typer =>
                 if (fd.tparams.nonEmpty)
                   err(msg"Type parameters are not yet supported in this position",
                     fd.tparams.head.toLoc)
+              
+              implicit val gl: GenLambdas =
```

Comment:
Let's not do that. The comment refers to a very specific call to `typeTerm`. There's no reason to type annotations with the same value of `GenLambdas`. For annotations, just use `false`.